File: easy/1614/1614.rb
# Runtime: 48 ms, faster than 94.92%
# Memory Usage: 209.9 MB, less than 64.41%
def max_depth(s)
  max = 0
  cnt = 0

  for i in (0...s.size) do 
    if s[i] == '('
      cnt += 1
    elsif s[i] == ')'
      cnt -= 1
    end

    max = (cnt > max) ? cnt : max
  end
  max
end

File: easy/1614/1614.cpp
/**
 * Runtime: 0 ms, faster than 100.00%
 * Memory Usage: 6.5 MB, less than 22.49% 
 */
class Solution {
  public:
    int maxDepth(string s) {
      int max = 0;
      int cnt = 0;

      for(int i=0; i<s.size(); ++i)
      {
        if (s[i] == '(') 
        {
          ++cnt;
        }
        else if (s[i] == ')')
        {
          --cnt;
        }

        if (cnt > max) 
          max = cnt;
      }

      return max;
    }
};
